### ISSUE TITLE
launch navigator use old api v2 instead of v3

### ISSUE THREAD

**Issue by petkoe** [driftyco/ionic-native#87]
Hello 

LN throws error using ionic-native 

launchnavigator.navigate() called using deprecated v2 API signature. Please update to use v3 API signature as deprecated API support will be removed in a future version

I think that is a problem using ionic native? 

Cordova CLI: 6.1.0 (cordova-lib@undefined)
Ionic Version: 2.0.0-beta.3
Ionic CLI Version: 2.0.0-beta.22
Ionic App Lib Version: 2.0.0-beta.12
ios-deploy version: 1.8.5 
ios-sim version: 5.0.7 
OS: Mac OS X El Capitan
Node Version: v5.7.0
Xcode version: Xcode 7.3 Build version 7D175 

cordova-plugin-actionsheet 2.2.2 "ActionSheet"
cordova-plugin-console 1.0.2 "Console"
cordova-plugin-device 1.1.1 "Device"
cordova-plugin-splashscreen 3.2.1 "Splashscreen"
cordova-plugin-statusbar 2.1.2 "StatusBar"
cordova-plugin-whitelist 1.2.1 "Whitelist"
ionic-plugin-keyboard 2.0.1 "Keyboard"
uk.co.workingedge.phonegap.plugin.launchnavigator 3.0.0 "Launch Navigator"

regards

peter

**Comment by ihadeed** [comment]
Is the plugin still functional? or it doesn't work at all?

I followed the GitHub docs while writing this plugin's wrapper, so it should work fine. I didn't notice that console warning before.

After reading the source code of the plugin, it seems like the developer has written an updated "API". It would take some time for us to implement the new API's as they do not have the `successCallback` and `errorCallback` in the main functions parameters anymore. They are expected inside the `options` parameter object.

We probably will end up doing minor modifications to Ionic Native to be compatible with the plugin's latest API.

**Comment by petkoe** [comment]
No it crashes both in emulator and on the device.

**Comment by ihadeed** [comment]
@petkoe could you provide the code you're using?

**Comment by petkoe** [comment]
Sure here it is

import {  App, IonicApp, Page, NavController, NavParams } from 'ionic-angular';

import {  Toast, LaunchNavigator } from 'ionic-native';
//  funktioniert gerade nicht in native
// InAppBrowser ist native 1.2.0

import * as mitgliederService from '../services/MitgliederService';

@Page({
    templateUrl: 'build/pages/details/details.html'
})

export class details {
    static get parameters() {
        return [[NavController], [NavParams], [Platform]];
    }

    constructor(nav, navParams, platform) {
        this.nav = nav;
        // If we navigated to this page, we will have an item available as a nav param
        this.mitglied = navParams.data;
       // var isRealDevice = ionic.Platform.isWebView();
    }


    NavigateContact( mitglied) {
        var ziel = this.mitglied.strasse +", " +this.mitglied.plz+" "+this.mitglied.ort;
        ziel = "Meraner Str 43, 86156 Augsburg, Germany";
        console.error( ziel );
		LaunchNavigator.navigate( ziel );
    }

    }


Basically i have list with street (strasse), zip code and city (ort) where i will give the user the possibility to navigate to it.

**Comment by ihadeed** [comment]
@petkoe I just did some quick debugging. The issue occurs when you don't supply the second optional parameter `start`. 

For now, you can pass `null` as the second parameter. This issue will be resolved in the next release.


```js
    LaunchNavigator.navigate( ziel , null );
```

**Comment by ihadeed** [comment]
Thanks Peter @petkoe

**Comment by petkoe** [comment]
Thanks to you Ibrahim. It now works thanks to your support. P
